demos/http/common/include/http_demo_utils.h

```diff
@@ -49,3 +52,60 @@ typedef int32_t ( * TransportConnect_t )( NetworkContext_t * pNetworkContext );
  */
 int32_t connectToServerWithBackoffRetries( TransportConnect_t connectFunction,
                                            NetworkContext_t * pNetworkContext );
+
+/**
+ * @brief Retrieve the path from the input URL.
+ *
+ * This function retrieves the location and length of the path from within the
+ * input the URL. The query is not included in the length returned.
+ *
+ * The URL MUST start with "http://" or "https://" to find the path.
+ *
+ * For example, if the URL is:
+ * pUrl = "https://www.somewebsite.com/path/to/item.txt?optionalquery=stuff"
+ *
+ * *pPath = "/path/to/item.txt?optionalquery=stuff"
+ * *pPathLen = 17
+ *
+ * @param[in] pUrl - URL string to parse.
+ * @param[in] urlLen - The length of the URL string input.
+ * @param[out] pPath - pointer within input url that the path starts at.
+ * @param[out] pPathLen - Length of the path.
```

Comment:
I don't think we ever add dashes. Does it have an effect on the doxygen output? A space should be fine.